pkg/resourcevalidator/common.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resourcevalidator
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	sharing "github.com/liqotech/liqo/apis/sharing/v1alpha1"
+	vkv1alpha1 "github.com/liqotech/liqo/apis/virtualkubelet/v1alpha1"
+)
+
+var cachelog = klog.LoggerWithName(klog.Background(), "[ webhook-cache ]")
+var shadowpodlog = klog.LoggerWithName(klog.Background(), "[ shadowpod-resource ]")
+var webhooklog = klog.LoggerWithName(klog.Background(), "[ webhook ]")
+var resourceofferlog = klog.LoggerWithName(klog.Background(), "[ resourceoffer-resource ]")
+var webhookrefreshlog = klog.LoggerWithName(klog.Background(), "[ webhook-refresh ]")
+
+// DecodeShadowPod decodes a shadow pod from a given runtime object.
+func (spv *ShadowPodValidator) DecodeShadowPod(obj runtime.RawExtension) (shadowpod *vkv1alpha1.ShadowPod, err error) {
+	shadowpod = &vkv1alpha1.ShadowPod{}
+	err = spv.decoder.DecodeRaw(obj, shadowpod)
+	return
+}
+
+func quotaFormatter(quota v1.ResourceList, quotaName string) string {
+	result := fmt.Sprintf("%s [ ", quotaName)
+	r := regexp.MustCompile("^hugepages-")
+	for k, v := range quota {
+		if res := r.MatchString(k.String()); !res {
+			result += fmt.Sprintf("%s: %s, ", k, v.String())
+		}
+	}
+	result += "]"
+	return result
+}
+
+/* func generateQuotaPattern(quota v1.ResourceList) v1.ResourceList {
+	quantity := resource.NewQuantity(0, resource.DecimalSI)
+	result := v1.ResourceList{}
+	for k := range quota {
+		result[k] = quantity.DeepCopy()
+	}
+	return result
+} */
+
+func getQuotaFromResourceOffer(resourceoffer *sharing.ResourceOffer) v1.ResourceList {
+	resources := v1.ResourceList{}
+
+	for key, value := range resourceoffer.Spec.ResourceQuota.Hard {
+		resources[key] = value
+	}
+	return resources
```

Comment:
Why not accessing the resources directly (`resourceoffer.Spec.ResourceQuota.Hard.DeepCopy()`)? Then, the function could also be dropped altogether.